api/v1/db/oracledb/parks-dao.js

```diff
@@ -0,0 +1,135 @@
+const appRoot = require('app-root-path');
+const _ = require('lodash');
+
+const conn = appRoot.require('api/v1/db/oracledb/connection');
+const { openapi } = appRoot.require('utils/load-openapi');
+const { serializeParks, serializePark } = require('../../serializers/parks-serializer');
+
+const getParameters = openapi.paths['/parks'].get.parameters;
+
+const sql = `
+  SELECT ID AS "id",
+  NAME AS "name",
+  STREET_ADDRESS AS "streetAddress",
+  CITY AS "city",
+  STATE AS "state",
+  ZIP AS "zip",
+  LATITUDE AS "latitude",
+  LONGITUDE AS "longitude",
+  OWNER_ID AS "ownerId",
+  BALLFIELD AS "ballfield",
+  BARBEQUE_GRILLS AS "barbequeGrills",
+  BASKETBALL_COURTS AS "basketballCourts",
+  BIKE_PATHS AS "bikePaths",
+  BOAT_RAMPS AS "boatRamps",
+  DOGS_ALLOWED AS "dogsAllowed",
+  DRINKING_WATER AS "drinkingWater",
+  FISHING AS "fishing",
+  HIKING_TRAILS AS "hikingTrails",
+  HORSESHOES AS "horseshoes",
+  NATURAL_AREA AS "naturalArea",
+  OFFLEASH_DOG_PARK AS "offleashDogPark",
+  OPEN_FIELDS AS "openFields",
+  PICNIC_SHELTERS AS "picnicShelters",
+  PICNIC_TABLES AS "picnicTables",
+  PLAY_AREA AS "playArea",
+  RESTROOMS AS "restrooms",
+  SCENIC_VIEW_POINT AS "scenicViewPoint",
+  SOCCER_FIELDS AS "soccerFields",
+  TENNIS_COURTS AS "tennisCourts",
+  VOLLEYBALL AS "volleyball"
+  FROM PARKS
+  WHERE 1 = 1
+`;
+
+// removes 'filter', '[', and ']' from parameter names to match sql column names
+const tidyKeyName = keyName => keyName.replace(/filter|\]|\[/gi, '');
+
+// converts amenities from query params to sql snippet
+const parseAmenities = (amenitiesArray, mode) => {
+  const sqlParams = _.reduce(amenitiesArray, (accumulator, value, index) => {
+    const enums = openapi.definitions.ParkResource.properties.attributes.properties.amenities.enum;
+    const sqlAmenity = _.snakeCase(amenitiesArray[index]).toUpperCase();
+    if (enums.includes(amenitiesArray[index])) {
+      if (index === 0) return `${sqlAmenity} = 1`;
+      return `${accumulator} ${mode === 'all' ? 'AND' : 'OR'} ${sqlAmenity} = 1`;
+    }
+    return undefined;
```

Comment:
Instead of returning `undefined` as the accumulator here, you should probably throw an exception or something. Ideally, this case should never occur since the query parameters are being validated by express-openapi